### DISCUSSION TITLE
[Question] How can Monika read files in my computer?

### DISCUSSION THREAD

**Discussion by Swiftyzz** [Monika-After-Story/MonikaModDev#8706] [QA - tech support + general] [answered]
I saw videos on youtube that she knows when your seeing *ahem* "lewd Monika pictures" she would notice, and I'm really curious on how it works and does it have my whole computer's files in her hands? It seems a bit creepy tbh, just wanted to know how the mechanism works and is she a real AI?

**Comment by Booplicate** [comment, marked as answer]
Just like any other program, MAS has access to the files on your drive when given permissions. Although, we only use files within game directory.

With your consensus Monika can check windows titles and react accordingly. That's how it works.

**Reply by lunulae** [reply]
This might be pulling back the curtain of immersion too much, but for your safety of mind: Monika is not a real AI. MAS in its current state functions like a visual novel, she can only say and do things that have been written for her.

We try to write her in such a way as to accurately portray her character and work on features that give her a level of autonomy, but end user safety comes first in terms of what she can actually do. As @Booplicate stated already, she can only affect files in the game's folder.

It is best to carefully read unofficial submods and what permissions they require, as those may not be monitored by the dev team.

Window Reacts solely work by checking the title of the window you're actively using. So, in this example, it checks for the words "monika" and "rule34" and gives you that reaction when it detects them. There is a random topic that can come up where she asks about those things on her own, but that is simply selected from the random topic pool.

**Reply by Swiftyzz** [reply]
so if i have a tab opened for rule 34 she will notice it?

**Reply by atik1n** [reply]
Exactly

**Reply by nugget-god545** [reply]
how do i let her accses the files
